share/spice/dota2/dota2.js

```diff
@@ -0,0 +1,190 @@
+(function (env) {
+    "use strict";
+
+    function getItemFromData(itemdata, itemName) {
+        var item;
+        if (itemdata[itemName]) {
+            item = itemdata[itemName];
+        } else {
+            $.each(itemdata, function(index, value) {
+                if (value.dname.toLowerCase() == itemName) {
+                    item = value;
+                }
+            });
+        }
+        return item;
+    }
+
+    function getItemComponents(item, fullResult) {
+        var components = [];
+        if (item.components) {
+            $.each(item.components, function(index, value) {
+                var component = {
+                    name: fullResult.itemdata[value].dname,
+                    img: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/items/' + fullResult.itemdata[value].img),
+                    searchTerms: fullResult.itemdata[value].dname.replace(" ", "+")
+                };
+                components.push(component);
+            });
+        }
+        return components;
+    }
+
+    function getItemDisplayAssets() {
+        return {
+            gold: {
+                image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/tooltips/gold.png'),
+                title: 'Gold'
+            }
+        }
+    }
+
+    function createItemReturnObject(item, result) {
+        return {
+            isItem: true,
+            name: item.dname,
+            description: item.desc,
+            cost: item.cost,
+            notes: item.notes,
+            attributes: item.attrib,
+            components: getItemComponents(item, result),
+            image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/items/' + item.img),
+            assets: getItemDisplayAssets()
+        };
+    }
+
+    function getHeroFromData(herodata, heroName) {
+        var hero;
+        if (herodata[heroName]) {
+            hero = herodata[heroName];
+            hero.heroKey = heroName;
+        } else {
+            $.each(herodata, function(index, value) {
+                if (value.dname.toLowerCase() == heroName) {
+                    hero = value;
+                    hero.heroKey = index;
+                }
+            });
+        }
+        return hero;
+    }
+
+    function getHeroRoles(hero) {
+        return hero.droles.replace(/ - /g, ', ');
+    }
+
+    function getHeroAbilities(hero, result) {
+        var abilities = [];
+        $.each(result.abilitydata, function(index, value) {
+            if (index.indexOf(hero.heroKey) === 0) {
+                abilities.push(
+                    {
+                        name: value.dname,
+                        image: DDG.getImageProxyURL('http://cdn.dota2.com/apps/dota2/images/abilities/' + index + '_hp1.png')
+                    }
+                );
+            }
+        });
+        return abilities;
+    }
+
+    function getHeroPrimaryAttribute(hero) {
```

Comment:
Seems like this could be replaced with an object lookup?

e.g. `primaryAttribute: primaryAttributes[hero.pa],`